File: 9041 - Software Instruction/test10/equi_filter.pl
#!/usr/bin/perl -w

@input = <STDIN>;

foreach $line (@input) {
	@array = split(' ',$line);
	my $line_count = -1;
	foreach $word (@array) {
		$line_count++;
		@chars = split('',$word);
		my $number = 0;
		my $count = -1;
		foreach $letter (@chars) {
			$count++;
			if ($#chars == 0) {
				if ($line_count == $#array) {
					print "$word";
				} else {
					print "$word ";	
				}
				#print "$word ";
				last;
			}
			if ($number == 0) {
				$number = findFrequency(\@chars,$letter);
			}
			if ($count == $#chars and $number == findFrequency(\@chars,$letter)) {
				if ($line_count == $#array) {
					print "$word";
				} else {
					print "$word ";	
				}
			}
			if ($number != findFrequency(\@chars,$letter)) {
				last;
			}
		}
	}
	print "\n";
}

sub findFrequency {
	my ($array_ref, $char) = @_;
	my @array = @{$array_ref};
	my $count = 0;
	#print "@array\n";
	foreach $letter (@array) {
		if (lc($letter) eq lc($char)) {
			$count++;
		}
	}
	return $count;
}

File: 9041 - Software Instruction/test10/equi_filter.py
#!/usr/bin/python3

import sys,re

def findFrequency(word_list,char):
	count = 0
	for i in word_list:
		if (char.lower() == i.lower()):
			count += 1
	return count

for line in sys.stdin.readlines():
	words_list = re.split('\s+',line)
	line_count = 0
	for word in words_list:
		line_count += 1
		number = 0;
		count = 0;
		for char in word:
			count += 1;
			if (len(word) == 1):
				if (line_count == len(words_list)):
					print(word, end='\n')
				else: 
					print(word, end=' ')
				break
				
			if (number == 0):
				number = findFrequency(word,char)
				
			if (count == len(word) and number == findFrequency(word,char)):
				if (line_count == len(words_list)):
					print(word, end='\n')
				else:
					print(word, end=' ')
			if (number != findFrequency(word,char)):
				break
	print("")
		